components/clp-py-utils/clp_py_utils/clp_metadata_db_utils.py

```diff
@@ -0,0 +1,140 @@
+) -> None:
+    db_cursor.execute(
+        f"""
+        CREATE TABLE IF NOT EXISTS `{archive_tags_table_name}` (
+            `archive_id` VARCHAR(64) NOT NULL,
+            `tag_id` INT unsigned NOT NULL,
+            PRIMARY KEY (`archive_id`,`tag_id`),
+            FOREIGN KEY (`archive_id`) REFERENCES `{archives_table_name}` (`id`),
+            FOREIGN KEY (`tag_id`) REFERENCES `{tags_table_name}` (`tag_id`)
+        )
+        """
+    )
+
+
+def _create_files_table(db_cursor, table_prefix: str) -> None:
+    db_cursor.execute(
+        f"""
+        CREATE TABLE IF NOT EXISTS `{table_prefix}{FILES_TABLE_SUFFIX}` (
+            `id` VARCHAR(64) NOT NULL,
+            `orig_file_id` VARCHAR(64) NOT NULL,
+            `path` VARCHAR(12288) NOT NULL,
+            `begin_timestamp` BIGINT NOT NULL,
+            `end_timestamp` BIGINT NOT NULL,
+            `num_uncompressed_bytes` BIGINT NOT NULL,
+            `begin_message_ix` BIGINT NOT NULL,
+            `num_messages` BIGINT NOT NULL,
+            `archive_id` VARCHAR(64) NOT NULL,
+            KEY `files_path` (path(768)) USING BTREE,
+            KEY `files_archive_id` (`archive_id`) USING BTREE,
+            PRIMARY KEY (`id`)
+        ) ROW_FORMAT=DYNAMIC
+        """
+    )
+
+
+def _create_column_metadata_table(db_cursor, table_name: str) -> None:
+    db_cursor.execute(
+        f"""
+        CREATE TABLE IF NOT EXISTS `{table_name}` (
+            `name` VARCHAR(512) NOT NULL,
+            `type` TINYINT NOT NULL,
+            PRIMARY KEY (`name`, `type`)
+        )
+        """
+    )
+
+
+def create_datasets_table(db_cursor, table_prefix: str) -> None:
+    """
+    Creates the dataset information table.
+
+    :param db_cursor: The database cursor to execute the table creation.
+    :param table_prefix: A string to prepend to the table name.
+    """
+    db_cursor.execute(
+        f"""
+        CREATE TABLE IF NOT EXISTS `{table_prefix}{DATASETS_TABLE_SUFFIX}` (
+            `name` VARCHAR(255) NOT NULL,
+            `archive_storage_directory` VARCHAR(4096) NOT NULL,
+            PRIMARY KEY (`name`)
+        )
+        """
+    )
+
+
+def create_metadata_db_tables(db_cursor, table_prefix: str, dataset: str | None = None) -> None:
+    """
+    Creates the standard set of tables for CLP's metadata.
+
+    :param db_cursor: The database cursor to execute the table creations.
+    :param table_prefix: A string to prepend to all table names.
+    :param dataset: If set, all tables will be named in a dataset-specific manner and the
+    column-metadata table will be created.
+    """
+    if dataset is not None:
+        table_prefix = f"{table_prefix}{dataset}_"
+
+    archives_table_name = f"{table_prefix}{ARCHIVES_TABLE_SUFFIX}"
+    tags_table_name = f"{table_prefix}{TAGS_TABLE_SUFFIX}"
+    archive_tags_table_name = f"{table_prefix}{ARCHIVE_TAGS_TABLE_SUFFIX}"
+    # TODO: Update the default column metadata table name after updating the indexer.
```

Comment:
```suggestion

    # TODO: Update this to
    # {table_prefix}{CLP_DEFAULT_DATASET_NAME}_{COLUMN_METADATA_TABLE_SUFFIX} when we can also
    # change the indexer to match.
```

* We try to include an empty line before comment blocks since it makes it a bit easier to read.
* I tried to make the TODO more actionable in case it's not you who addresses it.